### PR TITLE
Implement `useGamebananaModDownloadUrl`

### LINKED ISSUE #603: Implement `useGamebananaModDownloadUrl`
**Is your feature request related to a problem? Please describe.**
The 1-Click Install button currently always links to the home page.

**Describe the solution you'd like**
Implement [useGamebananaModDownloadUrl](../blob/main/src/hooks/gamebananaApi.ts#L252) so that the 1-Click Install button returns the same URL as the 1-Click Install button on GameBanana.

**Describe alternatives you've considered**
If this turns out not to be possible, we can just link to the mod's GameBanana page.

**Additional context**
Will need to reverse-engineer how the GameBanana URLs work. Follow-up to #563 item 5.

### PR DESCRIPTION
**Pull request by otobot1**
Closes #603

### CODE REVIEW

**Comment by otobot1** [comment]
@ShouvikGhosh2048 Would you mind reviewing this?

File: src/hooks/gamebananaApi.ts [hunk 1]
```diff
@@ -228,26 +228,136 @@ export const useGamebananaModImageUrls = (
...
+    "_sMd5Checksum": string,
+    "_sClamAvResult": string,
+    "_sAnalysisResult": string,
+    "_bContainsExe": boolean;
+};
+
+
+const isGamebananaFileMetadata = (
+    data: unknown
+): data is GamebananaFileMetadata => {
```
Comment (ShouvikGhosh2048) [inline src/hooks/gamebananaApi.ts:249]: I'm not familiar with this syntax, why is the `data is GamebananaFileMetadata` needed?

**Comment by ShouvikGhosh2048** [comment]
LGTM

File: src/hooks/gamebananaApi.ts (hunk 1 above)
Comment (otobot1) [reply src/hooks/gamebananaApi.ts:252]: It's a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). It's there to explicitly tell TypeScript "If this function returns true, the variable that was passed into the `data` parameter is *definitely* of type `GamebananaFileMetadata`. That way, after line 337 TypeScript can infer the correct type for `filesObject`.

**Pull request merged**